### PR TITLE
Feature/92 runs table string filter

### PR DESCRIPTION
**Pull request by jozefbakus**
implements #92
implements #89

### CODE REVIEW

File: ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts:1]: Please add some tests. E.g. you could subscribe to the `changes` subject and call `modelChanged` or emit values to `removeFilterSubject`

File: ui/src/app/components/runs/runs.component.html [hunk 2]
```diff
@@ -34,9 +34,17 @@
 
     <clr-dg-column [clrDgField]="dagRunColumns.WORKFLOW_NAME" [clrFilterValue]>
       <ng-container *clrDgHideableColumn="{hidden: false}">Workflow name</ng-container>
+      <clr-dg-filter [clrDgFilter]="workflowNameFilter">
+        <app-string-filter #workflowNameFilter [property]="dagRunColumns.WORKFLOW_NAME" [removeFiltersSubject]="removeFiltersSubject">
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/runs.component.html:39]: There is some duplication with `[property]` because it's already defined in the parent `clr-dg-column`. One way to resolve it is adding
```
  constructor(@Host() private column: ClrDatagridColumn) { }
```
to the `StringFilterComponent`. Then you can obtain the name from the column. However, obviously it adds a tight coupling to the `ClrDatagridColumn`, but one could argue that a filter without a column makes no sense

File: ui/src/app/components/runs/runs.component.ts [hunk 3]
```diff
@@ -62,18 +69,52 @@ export class RunsComponent implements OnDestroy, AfterViewInit {
...
+      size: this.pageSize,
+      sort: this.sort,
+      filters: filters
     };
 
     this.store.dispatch(new GetDagRuns(searchRequestModel));
   }
 
+  createFiltersModel(filters: {[prop:string]: any}): FiltersModel {
+    let byWorkflowNameOption = this.filters[dagRunColumns.WORKFLOW_NAME];
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/runs.component.ts]: You don't need `this.`

File: ui/src/app/components/runs/runs.component.ts [hunk 4]
```diff
@@ -62,18 +69,52 @@ export class RunsComponent implements OnDestroy, AfterViewInit {
...
+    let filters:{[prop:string]: any} = {};
+    if (state.filters) {
+      for (let filter of state.filters) {
+        let {property, value} = <{property: string, value: any}>filter;
+        filters[property] = value;
+      }
+    }
+    this.filters = filters;
+
+    this.refresh();
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/runs.component.ts:80]: You can pass filters directly into `this.refresh()`. Then you don't need `this.filters` at all.

File: ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts (hunk 1 above)
Comment (kevinwallimann) [reply ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts:1]: `accepts` also needs to be tested.

**Review by kevinwallimann — CHANGES_REQUESTED** [review]

File: ui/src/app/components/runs/runs.component.ts (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/components/runs/runs.component.ts:80]: Yes, I do need, because I call refresh method in view, where I don't have these values.

Comment (kevinwallimann) [reply ui/src/app/components/runs/runs.component.ts:80]: Ok I see

File: ui/src/app/components/runs/runs.component.ts (hunk 3 above)
Comment (kevinwallimann) [reply ui/src/app/components/runs/runs.component.ts]: Can you mark the method private?

Comment (kevinwallimann) [reply ui/src/app/components/runs/runs.component.ts]: Also, consider using `const` instead of `let`

Comment (jozefbakus) [reply ui/src/app/components/runs/runs.component.ts]: Fixed

File: ui/src/app/components/runs/runs.component.html (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/components/runs/runs.component.html:39]: Fixed

Comment (jozefbakus) [reply ui/src/app/components/runs/runs.component.html:39]: No, I've reverted it back. It is clearer this explicit way. It simplifies testing and I would like to avoid having references on parent elements from a child.

File: ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/runs/filters/string-filter/string-filter.component.spec.ts:1]: accepts has been tested

File: ui/src/app/components/runs/filters/string-filter/string-filter.component.ts [hunk 5]
```diff
@@ -32,20 +32,18 @@ export class StringFilterComponent implements ClrDatagridFilterInterface<DagRunM
   //clarity interface
   changes = new Subject<any>();
   //angular
-  modelChanges: Subject<string> = new Subject<string>();
+  modelChanges: Subject<any> = new Subject<any>();
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/filters/string-filter/string-filter.component.ts]: Why is this change required?

File: ui/src/app/components/runs/filters/string-filter/string-filter.component.ts [hunk 6]
```diff
@@ -69,7 +67,7 @@ export class StringFilterComponent implements ClrDatagridFilterInterface<DagRunM
 
   onRemoveFilter() {
     this.value = undefined;
-    this.modelChanges.next(undefined)
+    this.modelChanges.next(this.value)
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/filters/string-filter/string-filter.component.ts:70]: Why is this change needed?

Comment (jozefbakus) [reply ui/src/app/components/runs/filters/string-filter/string-filter.component.ts:70]: No reason. I just didnt want to repeat value.

Comment (kevinwallimann) [reply ui/src/app/components/runs/filters/string-filter/string-filter.component.ts:70]: Oh sorry, I completely missed `this.value = undefined;`

**Review by kevinwallimann — APPROVED** [review]
